gravity.c

```diff
@@ -0,0 +1,204 @@
+/* Pi-hole: A black hole for Internet advertisements
+*  (c) 2017 Pi-hole, LLC (https://pi-hole.net)
+*  Network-wide ad blocking via your own hardware.
+*
+*  FTL Engine
+*  Gravity database routines
+*
+*  This file is copyright under the latest version of the EUPL.
+*  Please see LICENSE file for your rights under this license. */
+
+#include "FTL.h"
+#include <sqlite3.h>
+
+// Private variables
+static sqlite3 *gravitydb = NULL;
+static sqlite3_stmt* stmt = NULL;
+
+// Prototypes from functions in dnsmasq's source
+void rehash(int size);
+
+// Open gravity database
+static bool gravityDB_open(void)
+{
+	struct stat st;
+	if(stat(FTLfiles.gravitydb, &st) != 0)
+	{
+		// File does not exist
+		logg("readGravity(): %s does not exist", FTLfiles.gravitydb);
+		return false;
+	}
+
+	int rc = sqlite3_open_v2(FTLfiles.gravitydb, &gravitydb, SQLITE_OPEN_READONLY, NULL);
+	if( rc ){
+		logg("readGravity() - SQL error (%i): %s", rc, sqlite3_errmsg(gravitydb));
+		sqlite3_close(gravitydb);
+		return false;
+	}
+
+	// Tell SQLite3 to store temporary tables in memory. This speeds up read operations on
+	// temporary tables, indices, and views.
+	char *zErrMsg = NULL;
+	rc = sqlite3_exec(gravitydb, "PRAGMA temp_store = MEMORY", NULL, NULL, &zErrMsg);
+	if( rc != SQLITE_OK ){
+		logg("readGravity(PRAGMA temp_store) - SQL error (%i): %s", rc, zErrMsg);
+		sqlite3_free(zErrMsg);
+		sqlite3_close(gravitydb);
+		return false;
+	}
+
+	return true;
+}
+
+// Close gravity database handle
+static void gravityDB_close(void)
```

Comment:
This is only used in two places, and one of those usages can be replaced by the other. Since this only has one statement and (soon to be only) one usage, it should be removed and the contents moved to where it's used.

Reply:
I agree, I removed this subroutine.